src/renderer/components/scroller.jsx

```diff
@@ -0,0 +1,91 @@
+/** @flow */
+import React, { Component, PropTypes } from 'react'
+import ReactDOM from 'react-dom'
+import shallowCompare from 'react-addons-shallow-compare'
+import raf from 'raf'
+
+export default class Scroller extends Component {
+  static propTypes = {
+    /**
+     * Function respondible for rendering children.
+     * This function should implement the following signature:
+     *
+     */
+    height: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.number]),
+    width: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.number]),
+    children: PropTypes.func.isRequired,
+
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      scrollTop: 0,
+      scrollHeight: 0,
+      offsetTop: 0,
+      scrollContainerHeight: 0
+    }
+
+    this._onScroll = this._onScroll.bind(this)
+  }
+
+  componentDidMount () {
+
+    this.setState({"scrollTop":this.refs.scrollContainer.scrollTop,
+      "scrollHeight":this.refs.scrollContainer.offsetHeight,
+      "offsetTop":ReactDOM.findDOMNode(this).getBoundingClientRect().top,
+      "scrollContainerHeight":ReactDOM.findDOMNode(this.refs.scrollContainer).getBoundingClientRect().height});
```

Comment:
Please break down the closing "tag" in a new line.
Also there is no need using quotes for these properties.